test/e2e/framework/secret/secret_helpers.go

```diff
@@ -0,0 +1,63 @@
+// +build e2e
+
+package secret
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/aad-pod-identity/test/e2e/framework"
+)
+
+// CreateInput is the input for Create.
+type CreateInput struct {
+	Creator      framework.Creator
+	Config       *framework.Config
+	Name         string
+	Namespace    string
+	ClientSecret string
```

Comment:
we could obtain the SP client secret from `config`, so we don't need an extra field to store `ClientSecret`

Reply:
The reason for keeping the `ClientSecret` is to allow each test to decide what they want to pass for `clientSecret`. This'll allow us to add negative test scenarios in the future if any. Lmk what you think. If we want to remove it for now I can make the change.

Reply:
SGTM